src/Game.js

```diff
@@ -0,0 +1,125 @@
+class Frame {
+  constructor () {
+    this._rolls = [];
+    this._bonus_score = 0;
+  }
+
+  isStrike() {
+    return (this._rolls[0] == 10);
+  }
+
+  isSpare() {
+    return (this._rolls[0] + this._rolls[1]== 10);
+  }
+
+  isFrameComplete() {
+    if (this.isStrike()) { return true; }
+    else if (this._rolls.length == 2 ) {return true;}
+  }
+
+  isFinalFrameComplete() {
+    if (this._rolls.length == 3) {return true;}
+  }
+
+  addRoll(score) {
+    this._rolls.push(score);
+  }
+  
+  addBonusScore(bonus) {
+    this._bonus_score += bonus;
+  }
+  
+  calcFrameTotal() {
+    return this._rolls.reduce((a, b) => a + b, 0) + this._bonus_score;
+  }
+
+  calcFrameTotalForFirstTwoRolls() {
+    return this._rolls.slice(0, 2).reduce((a, b) => a + b, 0);
+  }
+
+  firstRollValue() {
+    return this._rolls[0];
+  }
+}
+
+class frameFactory {
+  static createFrame() {
+     return new Frame();
+  }
+}
+
+class Game {
+  constructor () {
+    this.frames = [];
+    this.currentFrame = frameFactory.createFrame();
+  }
+
+  roll(score) {
+    this.currentFrame.addRoll(score);
+    if (this._isNewFrame()) {this._completeFrame();}
+  }
+
+  scoreTotal() {
+    let score = 0;
+    this.frames.forEach(frame => {
+      score += frame.calcFrameTotal();
+  });
+  return score;
+  }
+
+  _updateFrameTotals() {
+    if (this._isNotFirstFrame()) { this._addAnyBonuses();}
```

Comment:
I think the check to know whether it's the first frame or not would be better pushed down further towards where it's important to know that.
Here, this check is only important because you later try to know if the last frame was a spare or strike. If your `_isLastFrameStrike` and `_isLastFrameSpare` methods were more robust, they would not give an error when called while we are still on the first frame. Instead, they could just say false (if first frame, then last frame was not a spare/strike). This way, you save yourself this check here, and an extra method.

Reply:
If you do that, then you can simplify further, as your method `_updateFrameTotals` would then only be calling `this._addAnyBonuses()`, which means you could just call that directly, and remove the `_updateFrameTotals` method.